project/REVIEWING.md

```diff
@@ -0,0 +1,170 @@
+
+Ideally, documentation should reflect the expected behavior of the code.  No code review should
+take place in this step.
+
+There are no strict rules on the way a design is validated: we usually aim for a consensus,
+although a single maintainer approval is often sufficient for obviously reasonable changes. In
+general, strong disagreement expressed by any of the maintainers should not be taken lightly.
+
+Once design is approved, a maintainer should make sure to remove this label and add the next one.
+
+Possible transitions from this state:
+
+ * Close: design rejected
+ * `2-needs-code-review`: general case
+ * `3-needs-docs-review`: proposals with only documentation changes
+
+## Code review - `2-needs-code-review`
+
+Maintainers are expected to review the code and ensure that it is good quality and in accordance
+with the documentation in the PR.
+
+All tests should pass.
+
+Changes to code must be reviewed and approved (LGTM'd) by a minimum of two code maintainers. When
+the author of a PR is a maintainer, he still needs the approval of two other maintainers.
+
+Once code is approved according to the rules of the subsystem, a maintainer should make sure to
+remove this label and add the next one. If documentation is absent but expected, maintainers should
+ask for documentation and move to status `3-needs-docs-review` for docs maintainer to follow.
+
+Possible transitions from this state:
+
+ * Close
+ * `1-needs-design-review`: raises design concerns
+ * `3-needs-docs-review`: general case
+ * `4-ready-to-merge`: trivial change not impacting documentation
+
+## Docs review - `3-needs-docs-review`
+
+Maintainers are expected to review the documentation in its bigger context, ensuring consistency,
+completeness, validity, and breadth of coverage across all extent and new documentation.
+
+They should ask for any editorial change that makes the documentation more consistent and easier to
+understand.
+
+Changes and additions to docs must be reviewed and approved (LGTM'd) by a minimum of two docs
+sub-project maintainers. If the docs change originates with a docs maintainer, only one additional
+LGTM is required (since we assume a docs maintainer approves of their own PR).
+
+Once documentation is approved (see below), a maintainer should make sure to remove this label and
+add the next one.
+
+Possible transitions from this state:
+
+ * Close
+ * `1-needs-design-review`: raises design concerns
+ * `2-needs-code-review`: requires more code changes
+ * `4-ready-to-merge`: general case
+
+## Merge - `4-needs-merge`
+
+Maintainers are expected to merge this pull request as soon as possible.  They can ask for a
+rebase, or carry the pull request themselves.  These should be the easy PRs to merge.
+
+Possible transitions from this state:
+
+ * Merge: general case
+ * Close: carry PR
+
+When merging a pull request, the maintainer should consider applying one or multiple impact labels
```

Comment:
I think it would help if the list of "impact" labels were moved into this section (from below). I had a similar reaction when I read this section - it wasn't clear to me why I would need to add labels **after** a PR was merged, but seeing the list with the explanation for each one helped.